src/Microsoft.Azure.ServiceBus/Message.cs

```diff
@@ -1168,234 +232,26 @@ static void ValidatePartitionKey(string partitionKeyPropertyName, string partiti
-                this.ReplyToSessionId = originalMessage.ReplyToSessionId;
-            }
-
-            if (originalMessage.ContentType != null)
-            {
-                this.ContentType = originalMessage.ContentType;
-            }
-
-            foreach (KeyValuePair<string, object> property in originalMessage.Properties)
-            {
-                this.InternalProperties.Add(property);
-            }
-
-            // Publisher property is intended to be made public eventually.
-            // So it gets cloned even in client side cloning.
-            if (originalMessage.Publisher != null)
-            {
-                this.Publisher = originalMessage.Publisher;
-            }
-
-            // Copy any internal properties (which cannot be set through public API)
-            // only when it's not client side cloning (which is invoked using public API).
-            if (!clientSideCloning)
-            {
-                if ((originalMessage.InitializedMembers & MessageMembers.PartitionId) != 0)
-                {
-                    this.PartitionId = originalMessage.PartitionId;
-                }
-
-                ReceiverHeaders originalMessageReceiverHeaders = originalMessage.receiverHeaders;
-                if (originalMessageReceiverHeaders != null)
-                {
-                    // Don't copy LockToken or LockedUntilUtc
-                    this.BodyId = originalMessage.BodyId;
-                    this.DeliveryCount = originalMessageReceiverHeaders.DeliveryCount;
-                    this.SequenceNumber = originalMessageReceiverHeaders.SequenceNumber;
-                    this.EnqueuedTimeUtc = originalMessage.EnqueuedTimeUtc;
-                    this.EnqueuedSequenceNumber = originalMessageReceiverHeaders.EnqueuedSequenceNumber;
-                }
-
-                if ((originalMessage.initializedMembers & MessageMembers.DeadLetterSource) != 0)
-                {
-                    this.DeadLetterSource = originalMessage.DeadLetterSource;
-                }
-            }
-        }
-
-        /// <summary> Ensures that receiver headers. </summary>
-        void EnsureReceiverHeaders()
-        {
-            if (this.receiverHeaders == null)
-            {
-                this.receiverHeaders = new ReceiverHeaders();
-            }
-        }
-
-        /// <summary> Throw if disposed. </summary>
-        /// <exception cref="Fx.Exception"> Thrown when object disposed. </exception>
-        void ThrowIfDisposed()
-        {
-            if (this.disposed)
-            {
-                // TODO: throw Fx.Exception.ObjectDisposed("Message has been disposed.");
-                throw new ObjectDisposedException("Message has been disposed.");
-            }
-        }
-
-        /// <summary> Throw if not locked. </summary>
-        /// <exception cref="FxTrace.Exception"> Thrown when as error. </exception>
-        void ThrowIfNotLocked()
+        public sealed class SystemPropertiesCollection
```

Comment:
It's much easier to navigate to the class when the filename represent the class and not a container class. Not as critical as the name comment.